src/testdata/urlpatterntestdata.json

```diff
@@ -2282,5 +2361,103 @@
     "expected_match": {
       "protocol": { "input": "foobar", "groups": { "name": "foobar" }}
     }
+  },
+  {
+    "pattern": [{ "hostname": "bad hostname" }],
+    "expected_obj": "error"
+  },
+  {
+    "pattern": [{ "hostname": "bad#hostname" }],
+    "expected_obj": "error"
+  },
+  {
+    "pattern": [{ "hostname": "bad%hostname" }],
+    "expected_obj": "error"
+  },
+  {
+    "pattern": [{ "hostname": "bad/hostname" }],
+    "expected_obj": "error"
+  },
+  {
+    "skip": "TODO",
```

Comment:
rust-url bug I imagine? Is there a tracking bug?

Reply:
There doesn't seem to be any test with `\\:` in urltestdata, so I'm assuming that this either should not actually fail, or no one has noticed it before in rust-url.

Reply:
Ok. Can you change the skip reason to "likely bug in rust-url"

Reply:
Done